pageserver/src/layered_repository.rs

```diff
@@ -102,6 +103,21 @@ lazy_static! {
     .expect("failed to define a metric");
 }
 
+/// The name of the metadata file pageserver creates per timeline.
+pub const METADATA_FILE_NAME: &str = "metadata";
+
+/// All files and data, relevant to a certain pageserver timeline,
+/// found locally in its working directory.
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct LocalTimeline {
+    pub tenant_id: ZTenantId,
+    pub timeline_id: ZTimelineId,
+    pub has_metadata: bool,
+    pub layers: BTreeSet<LayerFileName>,
+    /// Latest Lsn that corresponds to the contents written to disk.
+    pub disk_consistent_lsn: Lsn,
```

Comment:
What is the value of `disk_consistent_lsn` and where do we get it if `has_metadata` is false?

Reply:
`disk_consistent_lsn` helps upload/download loop to determine, whether a certain timeline should be uploaded or not.
Currently, checkpointer thread always tries to freeze the layers on disk, disregarding on whether there were actual updates to the timeline.

We get it in two places:
* [during freezing](https://github.com/zenithdb/zenith/pull/686/files#diff-2dee9987054422607c1ab7369e102927d8477dea6d75c4c7f3891e3d65b15d1eR1467)
* [when loading timelines from disk during startup](https://github.com/zenithdb/zenith/pull/686/files#diff-2dee9987054422607c1ab7369e102927d8477dea6d75c4c7f3891e3d65b15d1eR1080)

Now I start to think, that the latter should be changed, good catch (luckily, only the WIP branch should be updated to fix this).
`has_metadata` case does not matter much for uploads/downloads, as it gets filtered out, but looks like we can throw it away in this struct entirely, replacing with `pub disk_consistent_lsn: Option<Lsn>`

Reply:
> we can throw it away in this struct entirely, replacing with `pub disk_consistent_lsn: Option<Lsn>`

Cool. That was exactly my thought.

Reply:
Implemented and pushed.

Reply:
> Implemented and pushed.

Hmm. I don't see the change... Looks like you missed to git add something?

Reply:
Quite a juggling with branches I've cast on myself, sorry.
Pushed now + here's the "real" change in the  WIP branch with better context on how that change is treated during uploads: https://github.com/zenithdb/zenith/pull/686/commits/66a5bdaa359890ddd3cfa036a96dff6640d511d9